lars-reimann/api-editor
# feat: improved filters in GUI

Closes #487.

### Summary of Changes

- [x] Replace setting "Show private declarations" with visibility filter
- [x] Replace `class:xy` etc. filter with separate filters `is:class` and `name:xy`
- [x] New filter: `annotation:any`
- [x] New filter: `annotation:@xy`
- [x] Negated filters
- [x] Display truly matching elements in bold (other elements in their hierarchy are still shown)
- [x] Update help text (popover to the right of the filter text box)
- [x] Bug: First parameter cannot be annotated during filtering even though it's not actually the first parameter
- [x] Unit tests for filters
- [x] Code documentation for filters

### Screenshots (if necessary)

![image](https://user-images.githubusercontent.com/2501322/170295223-c945d3ec-c467-4c2d-8ca0-7baa64943721.png)

### Testing Instructions

1. Open a terminal in the root of the repository folder
2. Run this command
```sh
cd api-editor/gui
npm run dev
```
3. Open [localhost:3000](http://localhost:3000) in the browser (if the port is taken it might be different, check the output of the previous command in your terminal)
3. Open the help dialog next to the filter box (top right corner)
4. Try different combinations of the filters listed there and observe the changes to the tree view.
